Assets/Mirror/Runtime/NetworkConnection.cs

```diff
@@ -375,9 +381,18 @@ internal bool InvokeHandler(int msgType, NetworkReader reader, int channelId)
         /// <returns></returns>
         public bool InvokeHandler<T>(T msg, int channelId) where T : IMessageBase
         {
+            // get writer from pool
+            NetworkWriter writer = NetworkWriterPool.GetWriter();
```

Comment:
we should be able to get rid of this completely,  InvokeHandler should be able to just call the handler with the message.   Future PR.